src/broker/Manager.cc

```diff
@@ -1026,10 +1026,11 @@ void Manager::ProcessStoreEventInsertUpdate(const TableValPtr& table, const std:
 
     const auto& its = table->GetType()->AsTableType()->GetIndexTypes();
     ValPtr zeek_key;
+    auto key_copy = key;
```

Comment:
I've changed the signature of `data_to_val` because it forced unnecessary copies. By taking a value, we always have to either produce a copy or `move` a value into place. By taking a reference, we're forcing the call side to make copies when necessary a priori and otherwise don't cause a copy. This makes the few instances where we actually need the copies visible now.

Ideally, `data_to_val` would take a `const broker::data&` reference. It's been a while, so I don't remember the details why I couldn't use that. It had something to do with `val_converter`. I'd like to clean that up eventually, but for now I at least wanted to get rid of some of the unnecessary copies.

Reply:
Got it — agreed.